### ISSUE TITLE
testcase test_trusted_context_connect.py failing with current clidriver

### ISSUE THREAD

**Issue by imavo** [ibmdb/python-ibmdb#944]
- Operating System Name: linux x64 (ubuntu 22.04 client , 20.04 Db2-server)
- db2level output from clidriver if in use: 11.5.9.0 special_36648, on client.
- Target Db2 Server Version: Db2-LUW 11.5.8.0
- Python Version: 3.10.2 (on client),  3.9.5 (on Db2-server)
- ibm_db version: 3.2.3
- For non-Windows, output of below commands:
  uname (Linux)
  uname -m (x86_64)
- Value of below environment variables if set:
  IBM_DB_HOME:  n/a
  PATH: (includes (...site-packages/*.egg/clidriver/bin)
  LIB/LD_LIBRARY_PATH/DYLD_LIBRARY_PATH: (...site-packages/.egg/clidriver/lib)

- Test script to reproduce the problem. (supplied with ibm_db)  test_trusted_context_connect.py
- For installation related issue, complete output of `pip install ibm_db` command. (not installation related)

Steps to Reproduce:

1. On linux x86_64, configure for running the existing `python ibmdb_tests.py` script, setup the users needed for the trusted-context tests and configure them in config.py and config.json etc. Create a python virtual-environment in which to test. Arrange to have two hostnames, a client (running python and ibm_db 3.2.3 with current clidriver), and a Db2-LUW server with configured test database and relevant userid/passwords already created and verified to work.
2. export SINGLE_PYTHON_TEST=test_trusted_context_connect.py
3. cd ...python-ibmdb-3.2.3
4. python ibm_db_tests
5. Resulting stdout is below: 
======================================================================
FAIL: test_trusted_context_connect (test_trusted_context_connect.IbmDbTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/user1/python-ibmdb-3.2.3/ibm_db_tests/test_trusted_context_connect.py", line 23, in test_trusted_context_connect
    obj.assert_expectf(self.run_test_trusted_context_connect)
  File "/home/user1/python-ibmdb-3.2.3/testfunctions.py", line 157, in assert_expectf
    self.assertNotEqual(result, None)
AssertionError: None == None

----------------------------------------------------------------------
Ran 1 test in 1.058s

FAILED (failures=1)

Investigation
==========
The immediate cause of the tests failing is that the expected output differs from the actual output.

That is to say in the testcase, the values listed below `#__LUW_EXPECTED__` do not match current 11.5.9.0 clidriver output.

In particular the reason code for SQL20361W when the switched-userid is not mentioned in the trusted-context definition, is now "1" when previously it was "2".  But there are other changes also that I suspect will require a new version of this testcase with adjusted __LUW_EXPECTED__ contents.

(narrative) When was the most recent time that the existing test-script "test_trusted_context_connect.py"
was run with a PASS (ok) for all tests, with a recent clidriver version (11.5.0.8 or higher) ?

Please can you run this testcase with a current-clidriver against Db2-LUW, because I believe the testcase needs adjusting (for the expected responses, as it appears that responses by clidriver have changed).

I ask because I observe some changes in clidriver behaviour for trusted-contexts on Db2-LUW, specifically for the case of SQL20361W when there is an attempt to switch to a userid that is not permitted by the trusted-context. This is one scenario tested by "test_trusted_context_connect.py".  There are other differences also.

Older versions of clidriver would return SQL20361W reason code "2" upon the next SQL-query after swtiching to an invalid userid+password (i.e. one not allowed by the trusted context). But current clidriver versions correctly return instead SQL20361W reason code "1" which is the reason code that matches the actual error.  

In addition, it seems that the 11.5.9.0 special_36648 clidriver (installed by default with ibm_db 3.2.3) will also drop the database-connection in the event of SQL20361W reason code "1" (which older clidriver versions did not do). I don't know if that is deliberate or accidental.

In short, please can you run this test test_trusted_context_connect.py in your environment with current clidriver against a current level of Db2-LUW server, and let me know if yours passes all tests without change.

**Comment by imavo** [comment]
I am closing this ticket.

The reason for closure is that I find that the testcase `test_trusted_context.py`  creates the trusted-context with only a single permitted ADDRESS (which is the value of config.hostname, which is that of the Db2-server to allow connections to succeed).  


That means when you run the testcase from remote (i.e. on a different hostname than the Db2-server) , the IP-address/Hostname is different from the permitted hostname in the trusted-context.  This causes the test failures.


It would be wiser for the test to be adjusted (along with the related `test_trusted_context_pconnect.py` ) so that the trusted context gets created with 2 addresses, one being that of the Db2-server (config.hostname) and the other being that of the current-hostname on which the test is executing (which can be different from the hostname of the Db2-server).


When I alter the testcase to __additionally__ include the address of the client-machine that actually runs the testcase (which is different from the Db2-server hostname), then I can get the tests to pass, or at least those tests that are correctly written and whose responses are exactly as expected.

Hence closing.
